# Compet - Formação 4.0

Aluna: Ruth Xavier

### Carregando o conjunto de dados

**Base escolhida:** Adequação da Formação Docente (2024)

**Descrição:** Percentual de docentes na Educação Infantil (Creche e Pré-Escola), no Ensino Fundamental, no Ensino Médio e Educação de Jovens e Adultos por grupo de adequação da formação à disciplina que leciona, segundo localização e dependência administrativa - Brasil, Regiões Geográficas e Unidades da Federação - 2024

#### Dicionário dos dados: 

01. NU_ANO_CENSO: Ano em que foi realizado o censo;
02. UNIDGEO: Unidade Geográfica; 
03. NO_CATEGORIA: Categorizando em zona Rural, Urbana e Total; 
04. NO_DEPENDENCIA: Dependência administrativa; 
05. ED_INF_CAT_1: Educação Infantil Grupo 1; 
06. ED_INF_CAT_2: Educação Infantil Grupo 2; 
07. ED_INF_CAT_3: Educação Infantil Grupo 3; 
08. ED_INF_CAT_4: Educação Infantil Grupo 5;
09. FUN_CAT_1: Ensino Fundamental Grupo 1 total;  
10. FUN_CAT_2: Ensino Fundamental Grupo 2 total;
11. FUN_CAT_3: Ensino Fundamental Grupo 3 total; 
12. FUN_CAT_4: Ensino Fundamental Grupo 4 total;  
13. FUN_CAT_5: Ensino Fundamental Grupo 5 total; 
14. FUN_AI_CAT_1: Ensino Fundamental Grupo 1 - Anos Iniciais; 
15. FUN_AI_CAT_2: Ensino Fundamental Grupo 2 - Anos Iniciais; 
16. FUN_AI_CAT_3: Ensino Fundamental Grupo 3 - Anos Iniciais; 
17. FUN_AI_CAT_4: Ensino Fundamental Grupo 4 - Anos Iniciais; 
18. FUN_AI_CAT_5: Ensino Fundamental Grupo 5 - Anos Iniciais; 
19. FUN_AF_CAT_1: Ensino Fundamental Grupo 1 - Anos Finais;
20. FUN_AF_CAT_2: Ensino Fundamental Grupo 2 - Anos Finais;
21. FUN_AF_CAT_3: Ensino Fundamental Grupo 3 - Anos Finais;
22. FUN_AF_CAT_4: Ensino Fundamental Grupo 4 - Anos Finais;
23. FUN_AF_CAT_5: Ensino Fundamental Grupo 5 - Anos Finais;
24. MED_CAT_1: Ensino Médio Grupo 1; 
25. MED_CAT_2: Ensino Médio Grupo 2;
26. MED_CAT_3: Ensino Médio Grupo 3;
27. MED_CAT_4: Ensino Médio Grupo 4;
28. MED_CAT_5: Ensino Médio Grupo 5;
29. EJA_FUN_CAT_1: Educação de Jovens e Adultos Grupo 1 - Ensino Fundamental;
30. EJA_FUN_CAT_2: Educação de Jovens e Adultos Grupo 2 - Ensino Fundamental;
31. EJA_FUN_CAT_3: Educação de Jovens e Adultos Grupo 3 - Ensino Fundamental;
32. EJA_FUN_CAT_4: Educação de Jovens e Adultos Grupo 4 - Ensino Fundamental;
33. EJA_FUN_CAT_5: Educação de Jovens e Adultos Grupo 5 - Ensino Fundamental;
34. EJA_MED_CAT_1: Educação de Jovens e Adultos Grupo 1 - Ensino Médio;
35. EJA_MED_CAT_2: Educação de Jovens e Adultos Grupo 2 - Ensino Médio;
36. EJA_MED_CAT_3: Educação de Jovens e Adultos Grupo 3 - Ensino Médio;
37. EJA_MED_CAT_4: Educação de Jovens e Adultos Grupo 4 - Ensino Médio;
38. EJA_MED_CAT_5: Educação de Jovens e Adultos Grupo 5 - Ensino Médio;
39. EJA_MED_CAT_1: Educação de Jovens e Adultos Grupo 1 - Ensino Médio;

GRUPO 1: Docentes com formação superior de licenciatura na mesma disciplina que lecionam, ou bacharelado na mesma disciplina com curso de complementação pedagógica concluído;

GRUPO 2: Docentes com formação superior de bacharelado na disciplina
correspondente, mas sem licenciatura ou complementação pedagógica;

GRUPO 3: Docentes com licenciatura em área diferente daquela que lecionam, ou com bacharelado nas disciplinas da base curricular comum e complementação pedagógica concluída em área diferente daquela que lecionam;

GRUPO 4: Docentes com outra formação superior não considerada nas categorias anteriores;

GRUPO 5: Docentes que não possuem curso superior completo.

In [1]:
# importando bibliotecas

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns 
from scipy.stats import skew, shapiro

ModuleNotFoundError: No module named 'matplotlib'

In [24]:
afd = pd.read_excel('datasets/AFD_BRASIL_REGIOES_UFS_2024.xlsx', skiprows=10)

In [25]:
afd.head()

NU_ANO_CENSO UNIDGEO NO_CATEGORIA NO_DEPENDENCIA ED_INF_CAT_1 ED_INF_CAT_2  \
0         2024  Brasil        Total          Total         64.1          0.7   
1         2024  Brasil       Urbana          Total         64.7          0.7   
2         2024  Brasil        Rural          Total         57.9          0.9   
3         2024  Brasil        Total        Federal         56.4          2.3   
4         2024  Brasil       Urbana        Federal         55.8          2.4   

  ED_INF_CAT_3 ED_INF_CAT_4 ED_INF_CAT_5 FUN_CAT_1  ... EJA_FUN_CAT_1  \
0         17.8          1.8         15.6      68.8  ...            27   
1         18.2          1.9         14.5      73.3  ...          35.5   
2         14.1          1.7         25.4      46.6  ...            10   
3         26.3          7.7          7.3      87.2  ...          68.9   
4         26.7          7.9          7.2      87.2  ...          68.9   

  EJA_FUN_CAT_2 EJA_FUN_CAT_3 EJA_FUN_CAT_4 EJA_FUN_CAT_5 EJA_MED_CAT_1  \
0           0.5          44.6           2.4          25.5          56.1   
1           0.7          47.9           2.7          13.2          57.9   
2           0.1          38.1           1.8            50          38.4   
3             6          17.5           5.5           2.1          83.3   
4             6          17.5           5.5           2.1          83.2   

  EJA_MED_CAT_2 EJA_MED_CAT_3 EJA_MED_CAT_4 EJA_MED_CAT_5  
0           1.1          36.1           3.1           3.6  
1           1.2          35.1           3.1           2.7  
2           0.6            46           3.6          11.4  
3           6.7           6.4           2.9           0.7  
4           6.8           6.3             3           0.7  

[5 rows x 39 columns]

In [26]:
afd.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 595 entries, 0 to 594
Data columns (total 39 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   NU_ANO_CENSO    594 non-null    object
 1   UNIDGEO         588 non-null    object
 2   NO_CATEGORIA    588 non-null    object
 3   NO_DEPENDENCIA  588 non-null    object
 4   ED_INF_CAT_1    588 non-null    object
 5   ED_INF_CAT_2    588 non-null    object
 6   ED_INF_CAT_3    588 non-null    object
 7   ED_INF_CAT_4    588 non-null    object
 8   ED_INF_CAT_5    588 non-null    object
 9   FUN_CAT_1       588 non-null    object
 10  FUN_CAT_2       588 non-null    object
 11  FUN_CAT_3       588 non-null    object
 12  FUN_CAT_4       588 non-null    object
 13  FUN_CAT_5       588 non-null    object
 14  FUN_AI_CAT_1    588 non-null    object
 15  FUN_AI_CAT_2    588 non-null    object
 16  FUN_AI_CAT_3    588 non-null    object
 17  FUN_AI_CAT_4    588 non-null    object
 18  FUN_AI_CAT

In [27]:
# Verificando a existência de dados nulos
afd.isna().any().any()

True

In [28]:
#observando o total de nulos por coluna
afd.isna().sum()

NU_ANO_CENSO      1
UNIDGEO           7
NO_CATEGORIA      7
NO_DEPENDENCIA    7
ED_INF_CAT_1      7
ED_INF_CAT_2      7
ED_INF_CAT_3      7
ED_INF_CAT_4      7
ED_INF_CAT_5      7
FUN_CAT_1         7
FUN_CAT_2         7
FUN_CAT_3         7
FUN_CAT_4         7
FUN_CAT_5         7
FUN_AI_CAT_1      7
FUN_AI_CAT_2      7
FUN_AI_CAT_3      7
FUN_AI_CAT_4      7
FUN_AI_CAT_5      7
FUN_AF_CAT_1      7
FUN_AF_CAT_2      7
FUN_AF_CAT_3      7
FUN_AF_CAT_4      7
FUN_AF_CAT_5      7
MED_CAT_1         7
MED_CAT_2         7
MED_CAT_3         7
MED_CAT_4         7
MED_CAT_5         7
EJA_FUN_CAT_1     7
EJA_FUN_CAT_2     7
EJA_FUN_CAT_3     7
EJA_FUN_CAT_4     7
EJA_FUN_CAT_5     7
EJA_MED_CAT_1     7
EJA_MED_CAT_2     7
EJA_MED_CAT_3     7
EJA_MED_CAT_4     7
EJA_MED_CAT_5     7
dtype: int64

Através da visualização dos dados com afd.head() e afd.info() e afd.isna().any().any() foi possível identificar que: 

- Há 39 colunas e 595 linhas; 
- A maioria das colunas possuem 588 valores que não são linhas, logo 6 linhas estão vazias (devido ao skiprow, geralmente exibe uma linha a mais no final); 
- E todas as colunas foram identificadas como object, pois provavelmente há valores não numéricos como o uso do "--" ou que estejam vazias.

## Preparação dos dados para análise

A base de dados referesse a uma coleta de dados realizada em 2024 a cerca do indiciador educacional a cerca da adequação da formação docente, logo isso significa que essa base tem por objetivo mapear se os docentes tem uma formação em conformidade entre a disciplina que lecionam e suas respectivas formações acadêmicas. 

De acordo com a LDB, Lei de Diretrizes e Base da Educação, Lei nº9.394/1996, no artigo 62, que contempla os dados presentes:

    Art. 62. A formação de docentes para atuar na educação básica far-se-á em nível superior, em curso de licenciatura, de graduação plena, em universidades e institutos superiores de educação, admitida, como formação mínima para o exercício do magistério na educação infantil e nos 5 (cinco) primeiros anos do ensino fundamental, a oferecida em nível médio na modalidade normal.

**Perguntas para os dados:**

Primeiramente, é importante definir um foco principal: Educação de Jovens e Adultos - Ensino Fundamental e Ensino Médio

*Educação de Jovens e Adultos* 

1. Em qual zona a EJA sofre uma maior discrepância entre a formação docente e a disciplina ensinada num total?

2. Analisando separadamente, qual zona apresenta maior discrepância na formação docente para a EJA do Ensino Fundamental? E para a EJA do Ensino Médio?

3. Em quais dependências admnistritativas há uma maior incidênica dessa discrepância em relação a EJA do Ensino Fundamental? E no Ensino Médio? 

4. Há alguma relação entre a unidade geográfica e a localização em realação a EJA do Ensino Fundamental?  E no Ensino Médio? 

5. Existe alguma correlação entre a unidade geográfica e a dependencia administrativa em relação à formação docente da EJA no Ensino Fundamental? E no Ensino Médio?

6. Ao fazer uma comparação entre o Ensino Fundamental EJA e Ensino Médio EJA, qual podemos observar que há uma maior dispariadade?

7. Qual região do Brasil apresenta a maior variação na formação dos docentes da EJA em termos de adequação (Grupos 1 a 5)?

8. Qual grupo de formação docente (1, 2, 3, 4 e 5) concentra um maior percentual desses docentes, considerando a zona geográfica?

9. Qual grupo de formação docente (Grupo 1, 2, 3, 4, 5) apresenta maior percentual de docentes na EJA, considerando a dependência administrativa?

10. Comparando as regiões do Brasil, qual apresenta maior desigualdade na adequação da formação docente para a EJA?


#### Selecionando dados  

As colunas selecionadas para a análise:

*possíveis filtros* 
UNIDGEO, NO_CATEGORIA, NO_DEPENDENCIA

*Etapa de ensino - Educação de Jovens e Adultos - Ensino Fundamental*
EJA_FUN_CAT_1, EJA_FUN_CAT_2, EJA_FUN_CAT_3, EJA_FUN_CAT_4, EJA_FUN_CAT_5

*Etapa de ensino - Educação de Jovens e Adultos - Ensino Médio*
EJA_MED_CAT_1,EJA_MED_CAT_2, EJA_MED_CAT_3, EJA_MED_CAT_4, EJA_MED_CAT_5


In [33]:
afd_eja = afd.iloc[:, [1, 2, 3, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38]]

afd_eja

UNIDGEO NO_CATEGORIA NO_DEPENDENCIA EJA_FUN_CAT_1 EJA_FUN_CAT_2  \
0    Brasil        Total          Total            27           0.5   
1    Brasil       Urbana          Total          35.5           0.7   
2    Brasil        Rural          Total            10           0.1   
3    Brasil        Total        Federal          68.9             6   
4    Brasil       Urbana        Federal          68.9             6   
..      ...          ...            ...           ...           ...   
590     NaN          NaN            NaN           NaN           NaN   
591     NaN          NaN            NaN           NaN           NaN   
592     NaN          NaN            NaN           NaN           NaN   
593     NaN          NaN            NaN           NaN           NaN   
594     NaN          NaN            NaN           NaN           NaN   

    EJA_FUN_CAT_3 EJA_FUN_CAT_4 EJA_FUN_CAT_5 EJA_MED_CAT_1 EJA_MED_CAT_2  \
0            44.6           2.4          25.5          56.1           1.1   
1            47.9           2.7          13.2          57.9           1.2   
2            38.1           1.8            50          38.4           0.6   
3            17.5           5.5           2.1          83.3           6.7   
4            17.5           5.5           2.1          83.2           6.8   
..            ...           ...           ...           ...           ...   
590           NaN           NaN           NaN           NaN           NaN   
591           NaN           NaN           NaN           NaN           NaN   
592           NaN           NaN           NaN           NaN           NaN   
593           NaN           NaN           NaN           NaN           NaN   
594           NaN           NaN           NaN           NaN           NaN   

    EJA_MED_CAT_3 EJA_MED_CAT_4 EJA_MED_CAT_5  
0            36.1           3.1           3.6  
1            35.1           3.1           2.7  
2              46           3.6          11.4  
3             6.4           2.9           0.7  
4             6.3             3           0.7  
..            ...           ...           ...  
590           NaN           NaN           NaN  
591           NaN           NaN           NaN  
592           NaN           NaN           NaN  
593           NaN           NaN           NaN  
594           NaN           NaN           NaN  

[595 rows x 13 columns]

In [ ]:
# Dicionários com as colunas selecionadas para análise

data_dictionary = pd.DataFrame([
    {
        'Variável': 'GEO',
        'Descrição': 'Unidade Geográfica',
        'Tipo': 'Categórico', 
        'Subtipo': 'Nominal',
        'Obs': 'Região do Brasil'
    },
    {
        'Variável': 'LOC',
        'Descrição': 'Categoria',
        'Tipo': 'Categórico', 
        'Subtipo': 'Nominal',
        'Obs': 'Zona Rural e Urbana'
    },
    {
        'Variável': 'DA',
        'Descrição': 'Dependência Administrativa',
        'Tipo': 'Categórico', 
        'Subtipo': 'Ordinal',
        'Obs': 'Municipal, Estudal, Federal e Privado'
    },
    {
        'Variável': 'EJAFUNG1',
        'Descrição': 'EJA - Ensino Fundamental - Grupo 1',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAFUNG2',
        'Descrição': 'EJA - Ensino Fundamental - Grupo 2',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAFUNG3',
        'Descrição': 'EJA - Ensino Fundamental - Grupo 3',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAFUNG4',
        'Descrição': 'EJA - Ensino Fundamental - Grupo 4',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAFUNG5',
        'Descrição': 'EJA - Ensino Fundamental - Grupo 5',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAMEDG1',
        'Descrição': 'EJA - Ensino Médio - Grupo 1',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAMEDG2',
        'Descrição': 'EJA - Ensino Médio - Grupo 2',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAMEDG3',
        'Descrição': 'EJA - Ensino Médio - Grupo 3',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAMEDG4',
        'Descrição': 'EJA - Ensino Médio - Grupo 4',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAMEDG5',
        'Descrição': 'EJA - Ensino Médio - Grupo 5',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    }
])

In [21]:
data_dictionary

Variável                           Descrição        Tipo   Subtipo  \
0        GEO                  Unidade Geográfica  Categórico   Nominal   
1         ZN                           Categoria  Categórico   Nominal   
2         DA          Dependência Administrativa  Categórico   Ordinal   
3   EJAFUNG1  EJA - Ensino Fundamental - Grupo 1    Numérico  Contínuo   
4   EJAFUNG2  EJA - Ensino Fundamental - Grupo 2    Numérico  Contínuo   
5   EJAFUNG3  EJA - Ensino Fundamental - Grupo 3    Numérico  Contínuo   
6   EJAFUNG4  EJA - Ensino Fundamental - Grupo 4    Numérico  Contínuo   
7   EJAFUNG5  EJA - Ensino Fundamental - Grupo 5    Numérico  Contínuo   
8   EJAMEDG1        EJA - Ensino Médio - Grupo 1    Numérico  Contínuo   
9   EJAMEDG2        EJA - Ensino Médio - Grupo 2    Numérico  Contínuo   
10  EJAMEDG3        EJA - Ensino Médio - Grupo 3    Numérico  Contínuo   
11  EJAMEDG4        EJA - Ensino Médio - Grupo 4    Numérico  Contínuo   
12  EJAMEDG5        EJA - Ensino Médio - Grupo 5    Numérico  Contínuo   

                                      Obs  
0                        Região do Brasil  
1                     Zona Rural e Urbana  
2   Municipal, Estudal, Federal e Privado  
3                                          
4                                          
5                                          
6                                          
7                                          
8                                          
9                                          
10                                         
11                                         
12

In [35]:
# renomeando as colunas 
colunas = ["GEO", "LOC", "DA", "EJAFUNG1", "EJAFUNG2", "EJAFUNG3", "EJAFUNG4", "EJAFUNG5", "EJAMED1", "EJAMED2", "EJAMED3", "EJAMED4", "EJAMED5"]
afd_eja.columns = colunas

afd_eja


GEO     LOC       DA EJAFUNG1 EJAFUNG2 EJAFUNG3 EJAFUNG4 EJAFUNG5  \
0    Brasil   Total    Total       27      0.5     44.6      2.4     25.5   
1    Brasil  Urbana    Total     35.5      0.7     47.9      2.7     13.2   
2    Brasil   Rural    Total       10      0.1     38.1      1.8       50   
3    Brasil   Total  Federal     68.9        6     17.5      5.5      2.1   
4    Brasil  Urbana  Federal     68.9        6     17.5      5.5      2.1   
..      ...     ...      ...      ...      ...      ...      ...      ...   
590     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
591     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
592     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
593     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
594     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   

    EJAMED1 EJAMED2 EJAMED3 EJAMED4 EJAMED5  
0      56.1     1.1    36.1     3.1     3.6  
1      57.9     1.2    35.1     3.1     2.7  
2      38.4     0.6      46     3.6    11.4  
3      83.3     6.7     6.4     2.9     0.7  
4      83.2     6.8     6.3       3     0.7  
..      ...     ...     ...     ...     ...  
590     NaN     NaN     NaN     NaN     NaN  
591     NaN     NaN     NaN     NaN     NaN  
592     NaN     NaN     NaN     NaN     NaN  
593     NaN     NaN     NaN     NaN     NaN  
594     NaN     NaN     NaN     NaN     NaN  

[595 rows x 13 columns]

In [36]:
afd_eja.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 595 entries, 0 to 594
Data columns (total 13 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   GEO       588 non-null    object
 1   LOC       588 non-null    object
 2   DA        588 non-null    object
 3   EJAFUNG1  588 non-null    object
 4   EJAFUNG2  588 non-null    object
 5   EJAFUNG3  588 non-null    object
 6   EJAFUNG4  588 non-null    object
 7   EJAFUNG5  588 non-null    object
 8   EJAMED1   588 non-null    object
 9   EJAMED2   588 non-null    object
 10  EJAMED3   588 non-null    object
 11  EJAMED4   588 non-null    object
 12  EJAMED5   588 non-null    object
dtypes: object(13)
memory usage: 60.6+ KB


É possível perceber os valores ausentes nessas 13 colunas seguem sendo identificadas como object. Portanto, para resolver esse problema irei substituir esses valores ausentes resolvi optar pela técnica de imputação Mediana, pela sua robustêz e por ser pouco afetada pelos outliers que podem sugir se eu usasse a média. 